EventView-1/src/main/java/com/eventview/repo/EventTypeRepoImpl.java

```diff
@@ -1,46 +1,81 @@
-import com.eventview.model.EvenTypes;
+import java.util.List;
 
 @Repository("eventTypeRepo")
-public class EventTypeRepoImpl implements EventTypeRepo{
-	
-	@Autowired
-	private JdbcTemplate jdbcTemplate;
-
-	@Override
-	public List<EvenTypes> getAllEvenTypes() {
-		String etsql = "select event_type_id,event_id,event_type from eventtypes";
-		return jdbcTemplate.query(etsql, new EvenTypeRowMapper());
-	}
-
-	@Override
-	public EvenTypes findByEventtypeId(Integer eventtypeid) {
-		String fbeti = "select * from eventtypes where event_type_id = ?";
-		return jdbcTemplate.queryForObject(fbeti, new Object[] { eventtypeid }, new EvenTypeRowMapper());
-	}
-
-	@Override
-	public void createEventType(EvenTypes evenTypes) {
-		jdbcTemplate.update(
-				"INSERT INTO eventtypes (event_type_id, event_id, event_type) VALUES (?,?,?)", evenTypes.getEventTypeId(), evenTypes.getEventid(), evenTypes.getEventType());
-	}
-
-	@Override
-	public void updateEventType(EvenTypes evenTypes) {
-		jdbcTemplate.update("update eventtypes set event_id=?, event_type=? where event_type_id=?", evenTypes.getEventid(), evenTypes.getEventType(), evenTypes.getEventTypeId());
-	}
-
-	@Override
-	public void deleteEventType(Integer eventtypeid) {
-		Object[] del = new Object[] {eventtypeid};
-		jdbcTemplate.update("delete from eventtypes where event_type_id=?", del);
-	}
+public class EventTypeRepoImpl implements EventTypeRepo {
+
+    private final Logger log = LoggerFactory.getLogger(EventRepoImpl.class);
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Override
+    public List<EvenTypes> getAllEvenTypes() {
+        String SELECT_ALL_EVENT_TYPE = "select event_type_id,event_type from eventtypes";
+        return jdbcTemplate.query(SELECT_ALL_EVENT_TYPE, new EvenTypeRowMapper());
+    }
+
+    @Override
+    public EvenTypes findByEventTypeId(Integer eventTypeId) {
+        try {
+            String SELECT_EVENT_TYPE_BY_ID = "select * from eventtypes where event_type_id = ?";
+            EvenTypes evenTypes = jdbcTemplate.queryForObject(SELECT_EVENT_TYPE_BY_ID, new Object[]{eventTypeId}, new EvenTypeRowMapper());
+            log.info("query generated " + SELECT_EVENT_TYPE_BY_ID + "-----" + eventTypeId);
+			return evenTypes;
+        } catch (Exception e) {
+            throw new EventTypeNotFoundException("EventType not found with id: "+ eventTypeId);
+        }
+    }
+
+    @Override
+    public void createEventType(EvenTypes evenTypes) {
+        String INSERT_EVENT_TYPE = "INSERT INTO eventtypes (event_type_id, event_type) VALUES (?,?)";
+        jdbcTemplate.update(INSERT_EVENT_TYPE, evenTypes.getEventTypeId(), evenTypes.getEventType());
+    }
+
+    @Override
+    public void updateEventType(EvenTypes evenTypes) {
+        String UPDATE_EVENT_TYPE = "update eventtypes set event_type=? where event_type_id=?";
+        jdbcTemplate.update(UPDATE_EVENT_TYPE, evenTypes.getEventType(), evenTypes.getEventTypeId());
+    }
```

Comment:
no exception handling, what if any error occurs while creating/updating

Reply:
will fix

Reply:
i'm checking existence of eventtype by using findbyeventtypeid(this handles exception) in controller